### PR TITLE
Close #1996 : use removeCache

### LINKED ISSUE #1996: destroyCache should use the "regular" removeCache method
in [EhcacheManager#683](https://github.com/ehcache/ehcache3/blob/e496850c8bf8191a30d7a783c80944c56d2cdc74/core/src/main/java/org/ehcache/core/EhcacheManager.java#L683) , ```destroyCache``` is calling ```removeAndCloseWithoutNotice``` that unfortunately does not trigger ```cacheRemoved```.

It can be problematic, since the user can destroy a cache programmatically without the monitoring capabilities knowing that a cache was indeed removed

### PR DESCRIPTION
**Pull request by anthonydahanne**

### CODE REVIEW

**Comment by anthonydahanne** [comment]
meh, test all threads failed
test this please

File: core/src/main/java/org/ehcache/core/EhcacheManager.java [hunk 1]
```diff
@@ -679,7 +679,7 @@ public void destroyCache(final String alias) throws CachePersistenceException {
     }
 
     try {
-      removeAndCloseWithoutNotice(alias);
+      removeCache(alias);
```
Comment (ljacomet) [inline core/src/main/java/org/ehcache/core/EhcacheManager.java]: should invoke `removeCache(alias, true);` instead

File: core/src/main/java/org/ehcache/core/EhcacheManager.java [hunk 2]
```diff
@@ -679,7 +679,7 @@ public void destroyCache(final String alias) throws CachePersistenceException {
     }
 
     try {
-      removeAndCloseWithoutNotice(alias);
```
Comment (ljacomet) [inline core/src/main/java/org/ehcache/core/EhcacheManager.java]: This method is orphaned by removing this call, please delete it.

File: core/src/main/java/org/ehcache/core/EhcacheManager.java (hunk 1 above)
Comment (ljacomet) [inline core/src/main/java/org/ehcache/core/EhcacheManager.java]: Not a chance this goes in without tests 😉 
And it needs daggy fix against `release/3.2`, `release/3.3` and `master`

**Comment by ljacomet** [comment]
@henri-tremblay from looking at history and trying to recover from memory, I believe there used to be stricter checks when removing a cache. And looking at the state of the code now, I don't see an issue using the same code and notifying indeed.

File: core/src/main/java/org/ehcache/core/EhcacheManager.java (hunk 2 above)
Comment (anthonydahanne) [inline core/src/main/java/org/ehcache/core/EhcacheManager.java]: yup

File: core/src/main/java/org/ehcache/core/EhcacheManager.java (hunk 1 above)
Comment (anthonydahanne) [inline core/src/main/java/org/ehcache/core/EhcacheManager.java]: oh, you mean ... we don't already test destroyCache() ? ;-)
I'll have a look. I'll also get a very deep breath for daggy fixing...

Comment (henri-tremblay) [inline core/src/main/java/org/ehcache/core/EhcacheManager.java]: Yes. But we are not testing that there are no notifications :-) Also, if in maintenance mode from destroyCache, it should stay there all the way through. Not sure this is tested as well.

**Comment by anthonydahanne** [comment]
replaced by #2004 #2005 #2006

**Pull request closed without merging**